### PR TITLE
Azure private endpoint support - follow-up

### PR DESCRIPTION
**Pull request by m1kola**
Addresses remaining points from this comment: https://github.com/openshift/hive/pull/1817#pullrequestreview-1043467943

### CODE REVIEW

File: pkg/remoteclient/remoteclient.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package remoteclient
 
-//go:generate mockgen -source=./remoteclient.go -destination=./mock/remoteclient_generated.go -package=mock
+//go:generate mockgen -destination=./mock/remoteclient_generated.go -package=mock  github.com/openshift/hive/pkg/$GOPACKAGE Builder
```
Comment (m1kola) [inline pkg/remoteclient/remoteclient.go:3]: Had to switch from source mode to reflect mode to be able to select a specific interface.

Comment (m1kola) [reply pkg/remoteclient/remoteclient.go]: Switched back to the source mode and moved dialer related code into separate files as per Eric's suggestion.

**Comment by 2uasimojo** [comment]
/lgtm

Thanks @m1kola

**Pull request merged**